packages/components/Button/src/useButton.ts

```diff
@@ -8,31 +8,68 @@ import { isHighContrast } from '@fluentui-react-native/theming-utils';
 
 import type { ButtonProps, ButtonInfo } from './Button.types';
 
+// On win32 we don't want to fire the onClick event if the Button
+// hasn't received a key down event first. This prevents behavior
+// like the button firing after you tab to it white Enter is pressed
+// and then releasing Enter, or the Menu reopening since it closes
+// onKeyDown while the Button operates onKeyUp.
+const shouldOnlyFireIfPressed = Platform.OS === ('win32' as any);
+
 export const useButton = (props: ButtonProps): ButtonInfo => {
   const defaultComponentRef = React.useRef(null);
   const {
-    onClick,
     accessible,
     accessibilityRole,
     componentRef = defaultComponentRef,
     disabled,
+    onBlur,
+    onClick,
+    onKeyDown,
     loading,
     enableFocusRing,
     focusable,
     ...rest
   } = props;
 
   const isDisabled = !!disabled || !!loading;
+  const [isKeyPressed, setIsKeyPressed] = React.useState<boolean>(false);
```

Comment:
\#supernitty wording could indicate relevant functionality e.g. keyboardInvocationStarted but not a big deal

Reply:
changed to isProcessingKeyboardInvocation